flutter_local_notifications/lib/src/platform_specifics/android/foreground_service_constants.dart

```diff
@@ -0,0 +1,45 @@
+/// Static helper class that contains all relevant
+/// Android service related constants.
+class AndroidServiceConstants {
```

Comment:
Yes, we could split them into to seperate things. For the start types we could use an enum, since these values are exclusive.
For the foreground service types an enum would work, too: Even if they are combinable, if we use a collection of them as you suggested below.

Reply:
Instead of regular enums though, I would suggest implementing via classes like https://github.com/MaikuB/flutter_local_notifications/blob/c373c5d3231423f89e40cef496e1420b2e88a994/flutter_local_notifications/lib/src/platform_specifics/android/enums.dart#L71. This is the way enums needed to be created before Dart supported them. The reason for this suggestion is that in case the Android APIs expand to include more types, a developer could manually create an instance of the appropriate class and pass the integer value they need

Reply:
While looking at this classes I noticed that they don't override `hashCode` and `==`. This will be necessary for the new foreground service type classe if instances should be used in a `Set` (see other discussion below). One could argue that this classes should only be referenced through the constants properties the respective class provides, but in my opinion the concept of equality clearly relates to the value they represent, so I would add `hashCode` and `==` based on the value. I would also make this addition to the already existing `Importance` and `Priority` classes if thats fine with you.

Reply:
Also, is there a reason why `Importance.value` us nullable?

Reply:
> I would also make this addition to the already existing Importance and Priority classes if thats fine with you.

Sure

> Also, is there a reason why Importance.value is nullable? Otherwise, I would change it.

Missed from doing null-safe migration but better to leave to do in another PR as that would be considered a breaking change